thorlcr/graph/thgraph.hpp

```diff
@@ -475,8 +476,10 @@ class graph_decl CGraphElementBase : public CInterface, implements IInterface
     unsigned getOutputs() const { return outputs.ordinality(); }
     bool isSource() const { return isActivitySource(kind); }
     bool isSink() const { return sink; }
-    inline bool doLogging() const { return log; }
+    inline bool doLogging() const { return log || detailedTracing; }
     inline void setLogging(bool _log) { log = _log; }
+    inline bool doDetailedTracing() const { return detailedTracing; }
+    inline void setDetailedTracing(bool _detailedTracing) { detailedTracing = _detailedTracing; }
```

Comment:
could we make this an unsigned so it can later be expanded to be a feature set at the activity level, e.g. for sort, to trace partitioning only, or merge only, etc. ?